### PR TITLE
copr: Add vectorized sha1

### PR DESCRIPTION
**Pull request by gauss1314**
Signed-off-by: gauss <gauss1314@gmail.com>

PCP #5751 

###  What have you changed?

Add vectorized sha1

Don't assume reviewers understand the original issue.

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by gauss1314** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/68

File: components/tidb_query/src/rpn_expr/impl_encryption.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+use openssl::hash::{self, MessageDigest};
+
+use crate::codec::data_type::*;
+use crate::codec::{Error, Result};
+
+#[rpn_fn]
+#[inline]
+pub fn sha1(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
+    Ok(match arg {
+        Some(arg) => Some(hex_digest(MessageDigest::sha1(), &arg).unwrap()),
```
Comment (rleungx) [inline components/tidb_query/src/rpn_expr/impl_encryption.rs]: I prefer not to unwrap here directly

**Comment by breeswish** [comment]
@rleungx PTAL again, thanks!

File: components/tidb_query/src/rpn_expr/impl_encryption.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use tidb_query_codegen::rpn_fn;
+
```
Comment (lonng) [inline components/tidb_query/src/rpn_expr/impl_encryption.rs]: Remove this empty line.

**Pull request merged**